test/specs/commonTests/implementsShorthandProp.tsx

```diff
@@ -18,34 +18,47 @@ export default Component => {
     const { mapsValueToProp } = options
     const { displayName } = ShorthandComponent
 
+    /**
+     * @param {ReactWrapper} wrapper [mounted Component]
+     * @param {(n: ReactWrapper) => boolean} propsCheckFn [boolean function that applies a boolean check on a mounted ShorthandComponent]
+     * return true if the ShorthandComponent that matches propsCheckFn is found
+     */
+    const testShorthandComponentProps = (
```

Comment:
this method is only used to check the following thing - essentially, to check that 
- there is a component in the rendered tree **of specific type**
- that has specific set of props being set
- and there is only one such component rendered

What if we would omit the predicate (that introduces generic use to the function, as it is not needed), and would just do it the following way

```ts
const expectContainsSingleShorthandElement = (wrapper, withProps) => ... // use ShorthandType here
```

Then it would be possible to use this function the following way for the test cases provided
```jsx
 test(`string value is handled as ${displayName}'s ${mapsValueToProp}`, () => {
   ...
   const shorthandProps = ...
   expectContainsSingleShorthandElement(mount(<Component {...props} />), shorthandProps)
}
```

Reply:
will address